examples/large_models/gpt_fast/handler.py

```diff
@@ -0,0 +1,191 @@
+import json
+import logging
+import time
+from pathlib import Path
+
+import torch
+from generate import _load_model, decode_one_token, encode_tokens, prefill
+from sentencepiece import SentencePieceProcessor
+
+from ts.handler_utils.timer import timed
+from ts.protocol.otf_message_handler import send_intermediate_predict_response
+from ts.torch_handler.base_handler import BaseHandler
+
+logger = logging.getLogger(__name__)
+
+
+class GptHandler(BaseHandler):
+    def __init__(self):
+        super().__init__()
+
+        self.model = None
+        self.tokenizer = None
+        self.context = None
+        self.prefill = prefill
+        self.decode_one_token = decode_one_token
+        self.initialized = False
+        self.device = torch.device("cpu")
+        self.prompt_length = 0
+
+    def initialize(self, ctx):
+        self.context = ctx
+        properties = ctx.system_properties
+        if torch.cuda.is_available() and properties.get("gpu_id") is not None:
+            self.map_location = "cuda"
+            self.device = torch.device(
+                self.map_location + ":" + str(properties.get("gpu_id"))
+            )
+
+        checkpoint_path = Path(ctx.model_yaml_config["handler"]["converted_ckpt_dir"])
+        assert checkpoint_path.is_file(), checkpoint_path
+
+        tokenizer_path = checkpoint_path.parent / "tokenizer.model"
+        assert tokenizer_path.is_file(), tokenizer_path
+
+        logger.info("Loading model ...")
+        t0 = time.time()
+        self.model = _load_model(checkpoint_path, self.device, torch.bfloat16, False)
```

Comment:
can we set "use_tp" by checking if "local_rank" in env, instead of hard coded as false?

Reply:
tp will be part of an update that will include continuous batching as well